src/removeUnusedPages.js

```diff
@@ -0,0 +1,85 @@
+const fs = require("fs");
+const path = require("path");
+
+/** @typedef {'nodejs' | 'dotnet' | 'java' | 'python'} LanguageFolder */
+
+/** @type {Record<LanguageFolder, string[]} */
+const languageExceptions = {
+  'nodejs': [
+    'mobile.mdx'
+  ]
+}
+
+const exceptions = [
+  'installation.mdx',
+  'showcase.mdx',
+]
+
+/**
+ * @param {LanguageFolder} language
+ * */
+function removeUnusedPages(languages) {
+  for(const language of languages) {
+    removeUnusedPageForLanguage(language)
+  }
+}
+
+function removeUnusedPageForLanguage(language) {
+  const languageFolder = path.join(__dirname, '..', language)
+  const sidebarPages = getSidebarPages(languageFolder)
+  const existingPages = readDirRecursive(path.join(languageFolder, 'docs'))
+  const unusedPages = getUnusedPages(language, languageFolder, sidebarPages, existingPages)
+  for(const page of unusedPages) {
+    console.log(`Removing unused page: ${page}`)
+    fs.rmSync(page)
+  }
+}
+
+function getSidebarPages(languageFolder) {
+  const sidebar = require(path.join(languageFolder, 'sidebars.js'))
+  const pages = []
+  const visit = (item) => {
+    if (item.type === 'doc') {
+      pages.push(path.join(languageFolder, 'docs', item.id + '.mdx'))
+    } else if (item.type === 'category') {
+      item.items.forEach(visit)
+    }
+  }
+  for (const key in sidebar)
+    sidebar[key].forEach(visit)
+  return pages
+}
+
+function readDirRecursive(dir) {
```

Comment:
You can use glob module here

Reply:
This would add another dependency (its already a peer-dependency) but still. Would like to keep this std-implementation.